tests/test_dependency_manager.py

```diff
@@ -0,0 +1,451 @@
+import os
+from typing import List
+from unittest import skipIf
+
+from django.test import override_settings
+from playwright.sync_api import Error, Page
+
+from django_components import types
+
+from .django_test_setup import setup_test_config
+from .testutils import PlaywrightTestCase
+
+IN_CI = os.environ.get("IN_CI", None)
+
+setup_test_config(
+    components={"autodiscover": False},
+    extra_settings={
+        "ROOT_URLCONF": "tests.test_dependency_manager",
+    },
+)
+
+urlpatterns: List = []
+
+
+class _BasePlaywrightTestCase(PlaywrightTestCase):
+    def _create_page_with_dep_manager(self) -> Page:
+        page = self.browser.new_page()
+        # Load the JS library by opening a page with the script, and then running the script code
+        # E.g. `http://localhost:54017/static/django_components/django_components.min.js`
+        script_url = self.live_server_url + "/static/django_components/django_components.min.js"
+        page.goto(script_url)
+        page.evaluate(
+            """() => {
+            eval(document.body.textContent);
+        }"""
+        )
+
+        # Ensure the body is clear
+        page.evaluate(
+            """() => {
+            document.body.innerHTML = '';
+            document.head.innerHTML = '';
+        }"""
+        )
+
+        return page
+
+
+@skipIf(IN_CI == "true", "Playwright tests currently cannot be run in CI")
+@override_settings(STATIC_URL="static/")
+class DependencyManagerTests(_BasePlaywrightTestCase):
+    def _create_page_with_dep_manager(self) -> Page:
+        page = self.browser.new_page()
+        # Load the JS library by opening a page with the script, and then running the script code
+        # E.g. `http://localhost:54017/static/django_components/django_components.min.js`
+        script_url = self.live_server_url + "/static/django_components/django_components.min.js"
+        page.goto(script_url)
+        page.evaluate(
+            """() => {
+            eval(document.body.textContent);
+        }"""
+        )
+
+        # Ensure the body is clear
+        page.evaluate(
+            """() => {
+            document.body.innerHTML = '';
+            document.head.innerHTML = '';
+        }"""
+        )
+
+        return page
+
+    def test_script_loads(self):
+        page = self._create_page_with_dep_manager()
+
+        # Check the exposed API
+        keys = sorted(page.evaluate("Object.keys(Components)"))
+        self.assertEqual(keys, ["createComponentsManager", "manager", "unescapeJs"])
+
+        keys = page.evaluate("Object.keys(Components.manager)")
+        self.assertEqual(
+            keys, ["callComponent", "registerComponent", "registerComponentData", "loadScript", "markScriptLoaded"]
+        )
+
+        page.close()
+
+
+# Tests for `manager.loadScript()` / `manager.markAsLoaded()`
+@skipIf(IN_CI == "true", "Playwright tests currently cannot be run in CI")
```

Comment:
As mentioned in one of the other comments, I had to skip the Playwright tests in CI. For that I defined a `IN_CI` env var in CI